lib/ansible/template/vars.py

```diff
@@ -57,6 +57,22 @@ def __init__(self, templar, globals, locals=None, *extras):
                     elif key not in ('context', 'environment', 'template'):
                         self._locals[key] = val
 
+    def __iter__(self):
+        for k in self._templar._available_variables:
```

Comment:
I think really want to inherit collections.Mapping like I've done here:

https://github.com/zmedico/ansible/pull/1/files